mkFit/MkFinder.cc

```diff
@@ -674,7 +679,15 @@ void MkFinder::AddBestHit(const LayerOfHits &layer_of_hits, const int N_proc,
     }
 
     bestHit[it] = -1;
-    minChi2[it] = m_iteration_params->chi2Cut;
+    const float invpt = Par[iP].At(it,3,0);
+    const float theta = std::fabs(Par[iP].At(it,5,0)-Config::PIOver2);
+    float this_c2 = (ILC.c_c2_0)*invpt+(ILC.c_c2_1)*theta+(ILC.c_c2_2);  
+    // In case layer is missing (e.g., seeding layers, or too low stats for training), leave original limits
+    if(this_c2>10.f){
+      max_c2 = (ILC.c_c2_sf)*this_c2;
+    }
+    else max_c2=10.0;
+    minChi2[it] = max_c2;
```

Comment:
Can we put this in a member function and use it in all 4 places this is cut-n-pasted? I guess you only need to pass `it` and maybe `iP`.
I'd implement it in .cc file and mark it inline.
Then you probably also don't need `ILC` variable and you can just declare `max_c2` when you call this function.

Reply:
Sure, will do and push

Reply:
Please, check. I have created a function which is similar to gitHitSelDynamicWindows, for consistency. As also suggested by @cerati I have made few other changes, in order to replace chi2Cut parameter with chi2Cut_min everywhere, and use it instead of a hardcoded value of 10 as min chi2 cut.

Reply:
I'd have the function return float and remove some of the local vars. I can push the change to the branch, if you're ok with it.

Reply:
No objection on my side.

Reply:
Done ... now you have to check :)

Reply:
Looks all good to me, thanks!